src/quadrature/quadrature_nodal_3D.C

```diff
@@ -103,11 +104,21 @@ void QNodal::init_3D(const ElemType, unsigned int)
     case TET10:
     case PRISM18:
     case HEX27:
+    case PYRAMID13:
+    case PYRAMID14:
       {
         QSimpson rule(/*dim=*/3, /*ignored*/_order);
         rule.init(_type, /*ignored*/_p_level);
         _points.swap (rule.get_points());
         _weights.swap(rule.get_weights());
+
+        // We can't do a proper Simpson rule for pyramids regardless
```

Comment:
Not sure I understand this, creating a `QSimpson` rule for a `PYRAMID13` a few lines prior to this will have already thrown a "Unsupported type" exception, so you should never get here...?

Reply:
We don't throw "Unsupported type" there anymore, we just return the nodal points (plus the 14th) with the dubious only-integrate-linears-exactly weights.